c/tskit/trees.c

```diff
@@ -2668,21 +2668,20 @@ tsk_treeseq_two_locus_count_stat(const tsk_treeseq_t *self, tsk_size_t num_sampl
         ret = TSK_ERR_MULTIPLE_STAT_MODES;
         goto out;
     }
-    if (state_dim < 1) {
-        ret = TSK_ERR_BAD_STATE_DIMS;
-        goto out;
-    }
     // TODO: impossible until we implement branch/windows
     // if (result_dim < 1) {
     //     ret = TSK_ERR_BAD_RESULT_DIMS;
     //     goto out;
     // }
-
     ret = tsk_treeseq_check_sample_sets(
         self, num_sample_sets, sample_set_sizes, sample_sets);
     if (ret != 0) {
         goto out;
     }
+    if (state_dim < 1) {
+        ret = TSK_ERR_BAD_STATE_DIMS;
+        goto out;
+    }
```

Comment:
I reordered this to avoid a cryptic error about state dims. If a user passes in empty sample sets, they should get an error about empty sample sets instead of a "BAD_STATE_DIM" error

Reply:
In fact, I'm not even sure how this could be triggered, since the state dim is just `num_sample_sets`. I think this could be removed, what do you think?